src/togamgxs/main.py

```diff
@@ -0,0 +1,171 @@
+# Optimization paramenters: user input required
+
+from collections import OrderedDict
+
+import openmc
+import pyrankvote
+from pyrankvote import Candidate, Ballot
+import matplotlib.pyplot as plt
+import pdb
+
+from togamgxs.getfilename import FileName
+from togamgxs.mgxs import MGXS
+from togamgxs.yakxs import YAKXS
+
+
+
+def TOGA(mgxs_types, group_structure_info, legendre_order, execute=True, optimize=False,
+         quantify_error=False, calc_equivalence=False, MPI=False,
+         LibraryName="MGXS_library", Description="", Generator="Self", TimeCreated=None):
+
+    equivalence = calc_equivalence
+    legendre = legendre_order
+
+    if optimize:
+        opt_strategy = optimize['opt_strategy']
+        opt_tolerance = optimize['opt_tolerance']
+        iso_histogram = optimize['iso_histogram']
+        angle_histogram = optimize['angle_histogram']
+        quantify_error = False
+
+    if MPI:
+        machinefile = MPI['machinefile']
```

Comment:
Same question as above, `MPI` is a boolean, so it can't also be used as a dictionary to define `machinefile`, etc.